eng/cibuild_bootstrapped_msbuild.sh

```diff
@@ -58,8 +58,10 @@ bootstrapRoot="$Stage1Dir/bin/bootstrap"
 
 if [ $host_type = "core" ]
 then
-  _InitializeBuildTool="$_InitializeDotNetCli/dotnet"
-  _InitializeBuildToolCommand="$bootstrapRoot/net8.0/MSBuild/MSBuild.dll"
+  _InitializeBuildTool="$bootstrapRoot/net8.0/dotnet"
+
+  # Must be consistent with the version in BootStrapMsBuild.props
+  _InitializeBuildToolCommand="$bootstrapRoot/net8.0/sdk/8.0.302/MSBuild.dll"
```

Comment:
Similar here - if possible, it'd be nice to have a sinlge variable in single place for the repo. If not easily possible - let's have it documented in retargetting checklist (since such checklist doesn't exist yet, it's fine to have just initial stub, which will have instructions just for bootstrap for now, and during next retargetting: https://github.com/dotnet/msbuild/issues/10280, we should ammend other steps)